### PR TITLE
[TT-9904] address cves

### PR DESCRIPTION
**Pull request by jeffy-mathew**
## Description
This PR addresses CVEs on tyk gateway docker image. 

* not upgrading python3.9 since it'll be updated in the next debian patch
* upgrading logstashhook dependency

Updated dependencies
- github.com/bshuster-repo/logrus-logstash-hook v0.4.1 => v1.1.0
- github.com/miekg/dns v1.0.14 => v1.1.56
- github.com/pires/go-proxyproto v0.0.0-20190615163442-2c19fd512994 => v0.7.0

## Related Issue
https://tyktech.atlassian.net/browse/TT-9904

## Screenshots (if appropriate)
Screenshot 2023-09-29 at 10 14 54 AM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

File: gateway/server.go [hunk 1]
```diff
@@ -1154,25 +1154,23 @@ func (gw *Gateway) setupLogger() {
...
 			conn, err = gas.Dial(gwConfig.LogstashTransport, gwConfig.LogstashNetworkAddr)
-			if err == nil {
-				hook, err = logstashHook.NewHookWithConn(conn, appName)
-			}
 		}
 
 		if err != nil {
 			log.Errorf("Error making connection for logstash: %v", err)
 		} else {
+			hook = logstashHook.New(conn, logstashHook.DefaultFormatter(logrus.Fields{
```
Comment (jeffy-mathew) [inline gateway/server.go:1171]: This will change the timestamp format to be `RFC3339Nano`

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5556/merge
Commit:  
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6347253381)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5556/merge
Commit: e1330ad3c8a0583edbbaa4d384a613ee8c16f9dd 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6347624265)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5556/merge
Commit: 501abd464caf1808e67837a768128aa125b0374c 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6388882555)

**Review by titpetric — APPROVED** [review]

**Pull request merged**